src/container_type.rs

```diff
@@ -0,0 +1,84 @@
+use k8s_openapi::{
+    api::core::v1::ResourceRequirements, apimachinery::pkg::api::resource::Quantity,
+};
+
+/// Describes the type of container. The types are:
+///
+/// - `Main`: This type of container runs the main appliction, like Superset or
+///    NiFi
+/// - `Init`: This type of container runs only once at the start of the product
+///    cluster and executes initial tasks like database migrations or insertion
+///    of data
+/// - `Sidecar`: Sidecar containers run alongside the main containers for
+///    additional tasks
+#[derive(Clone, Debug, Default)]
+pub enum ContainerType {
+    #[default]
+    Main,
+    Init,
+    Sidecar,
+}
+
+impl From<ContainerType> for ResourceRequirements {
+    fn from(container_type: ContainerType) -> Self {
+        Self::from(&container_type)
+    }
+}
+
+impl From<&ContainerType> for ResourceRequirements {
```

Comment:
We really only want to set default resources on init containers

Reply:
Yeah makes sense!